src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementFhirImportService.cs

```diff
@@ -90,9 +120,85 @@ public async Task ProcessEventsAsync(IEnumerable<IEventMessage> events, string t
                 }
             }
 
-            // Step 3: Transform the Measurement Groups into Observations and write to the FHIR Server.
-            //         Exceptions are currently handled in ProcessMeasurementGroups
-            await ProcessMeasurementGroups(measurementGroups, template, log, measurementToEventMapping).ConfigureAwait(false);
+            // Transform the Measurement Groups into Observations and write to the FHIR Server.
+            // Exceptions are currently handled in SendMeasurementGroups
+            await SendMeasurementGroups(measurementGroups, template, log, measurementToEventMapping).ConfigureAwait(false);
+        }
+
+        private async Task ProcessMeasurementGroupEvents(IEnumerable<IEventMessage> events, ILookupTemplate<IFhirTemplate> template, ITelemetryLogger log)
+        {
+            if (events.Count() == 0)
+            {
+                return;
+            }
+
+            foreach (var evt in events)
+            {
+                try
+                {
+                    IEnumerable<MeasurementGroup> groupedMeasurements;
+
+                    try
+                    {
+                        var partitionId = events.FirstOrDefault()?.PartitionId;
+
+                        // decompress the measurement group if it is compressed
+                        byte[] bytes;
+                        if (evt.BodyContentType == "application/gzip")
+                        {
+                            bytes = Compression.DecompressWithGzip(evt.Body.ToArray());
+                            evt.Body = new ReadOnlyMemory<byte>(bytes);

```

Comment:
It is technically being used in the case where an exception is being thrown. If an exception is thrown I wanted the error message service to have access to the uncompressed bytes rather than the compressed bytes. But the downside is that we are setting this evt.Body on every message whether an exception is thrown or not. We could change this to have the error message service deal with the uncompressing if that is preferred

Reply:
I kind of like that approach but if we do that then we would want separate the responsibility for parsing the message into it's own class so we can reuse the logic.

Reply:
I removed it for now and will add a work item for a refactor for the error message service